public/js/maho-captcha.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Maho
+ *
+ * @package     Maho_Captcha
+ * @copyright   Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+const MahoCaptcha = {
+    loading: null,
+    altchaWidget: null,
+    altchaState: null,
+    frontendSelectors: '',
+    onVerifiedCallback: null,
+
+    async setup(config) {
+        this.altchaWidget = document.querySelector('altcha-widget');
+        this.frontendSelectors = config.frontendSelectors ?? '';
+        this.loading = config.loading ?? '';
+
+        if (document.readyState === 'loading') {
+            document.addEventListener('DOMContentLoaded', this.initForms.bind(this));
+        } else {
+            this.initForms();
+        }
+    },
+
+    initForms() {
+        for (const formEl of document.querySelectorAll(this.frontendSelectors)) {
+            formEl.addEventListener('focusin', this.loadAltchaScripts.bind(this), true);
+            formEl.addEventListener('submit', this.onFormSubmit.bind(this), true);
+        }
+    },
+
+    async loadAltchaScripts() {
+        // Load Altcha JS
+        await Promise.all([
+            new Promise((resolve, reject) => {
+                const script = document.createElement('script');
+                script.src = '/js/altcha.min.js';
+                script.type = 'module';
+                script.onload = resolve;
+                script.onerror = () => reject(`${script.src} Not Found`);
+                document.head.appendChild(script);
+            }),
+        ]);
+
+        // Inject Stylesheet
+        const styleEl = document.createElement('style');
```

Comment:
Same with this check: 
```js
if (!document.querySelector('style[maho-captcha-style]')) {
```

Reply:
what about this last version?

Reply:
I'd say that altcha.min.js will always be loaded via loadAltchaScripts() so we should be safe there :-)